openjdk/build.xml

```diff
@@ -77,103 +77,132 @@
 			<else>
 				<if>
-					<and>
-						<contains string="${env.SPEC}" substring="aarch64" />
-						<equals arg1="${JDK_VERSION}" arg2="8" />
-					</and>
-  						<then>
-							<property name="jdkName" value="openjdk-aarch64-jdk8u" />
-  						</then>		
-					<else>
+					<matches string="${JDK_IMPL}" pattern="openj9|ibm" />
+					<then>
 						<if>
-							<or>
-								<equals arg1="${JDK_VERSION}" arg2="8" />
-								<equals arg1="${JDK_VERSION}" arg2="9" />
-								<equals arg1="${JDK_VERSION}" arg2="10" />
-								<equals arg1="${JDK_VERSION}" arg2="11" />
-								<equals arg1="${JDK_VERSION}" arg2="12" />
-								<equals arg1="${JDK_VERSION}" arg2="13" />
-                                                                <equals arg1="${JDK_VERSION}" arg2="14" /> 
-							</or>
+							<equals arg1="${env.ORIGIN_JDK_VERSION}" arg2="next" />
 							<then>
-								<property name="jdkName" value="openjdk-jdk${JDK_VERSION}u" />
+								<property name="jdkName" value="openj9-openjdk-jdk" />
 							</then>
 							<else>
-								<property name="jdkName" value="openjdk-jdk${JDK_VERSION}" />
+								<if> <contains string="${SPEC}" substring="zos"/>
+									<then>
+										<property name="jdkName" value="openj9-openjdk-jdk${JDK_VERSION}-zos" />
+									</then>
+									<else>
+										<property name="jdkName" value="openj9-openjdk-jdk${JDK_VERSION}" />
+									</else>
+								</if>
+							</else>
+						</if>
+						<if>
+			                                    <contains string="${SPEC}" substring="zos"/>
+			                                    <then>
+								<property name="regressionRepo" value="git@github.ibm.com:runtimes/${jdkName}.git" />
+			                                    </then>
+			                                    <else>
+						        	<property name="regressionRepo" value="https://github.com/ibmruntimes/${jdkName}.git" />
 							</else>
 						</if>
-					</else>
-				</if>
-				<property name="regressionRepo" value="https://github.com/AdoptOpenJDK/${jdkName}.git" />
-			</else>
-		</if>
-		<if>
-			<isset property="env.RELEASE_TAG"/>
-			<then>
-				<echo message="git clone --depth 1 -q -b ${env.RELEASE_TAG} ${regressionRepo}" />
-				<exec executable="git" failonerror="true">
-					<arg line="clone --depth 1 -q -b ${env.RELEASE_TAG} ${regressionRepo}" />
-				</exec>
-				<move file="${jdkName}" tofile="openjdk-jdk"/>
-			</then>
-			<else>
-				<if>
-					<contains string="${JDK_IMPL}" substring="hotspot"/>
-					<then>
-						<property name="defaultBranch" value="-b dev"/>
 					</then>
 					<else>
-						<property name="defaultBranch" value=""/>
+						<if>
+							<and>
+								<contains string="${env.SPEC}" substring="aarch64" />
+								<equals arg1="${JDK_VERSION}" arg2="8" />
+							</and>
+								<then>
+									<property name="jdkName" value="openjdk-aarch64-jdk8u" />
+								</then>		
+							<else>
+								<if>
+									<or>
+										<equals arg1="${JDK_VERSION}" arg2="8" />
```

Comment:
Would be good if we could escape this delicate handling of JDK_VERSION, that needs continual revision...  I know its been attempted in the past, but wonder if we can just check first for existence of a openjdk-jdk${JDK_VERSION}u repo and use it, else check for openjdk-jdk {$JDK_VERSION} repo and use it else exit in error.

Reply:
I think we could try `git ls-remote` 

```
git ls-remote -h https://github.com/AdoptOpenJDK/openjdk-jdk19999u.git
remote: Repository not found.

git ls-remote -h https://github.com/AdoptOpenJDK/openjdk-jdk11u.git
7d85e23ca2cb69fc3a2702e8ccabbbd0de8a6dfa	refs/heads/archived-dev
824f8474f53334f4f5e4f30e1ba3c7de42666b27	refs/heads/dev
5fb9ef7739a4cab3574e8817d1c46102b8871349	refs/heads/master
8a9a4983b964f241a563e27569726d7cf77860ef	refs/heads/release
```

https://superuser.com/questions/227509/git-ping-check-if-remote-repository-exists

Reply:
Interesting I tried with non-exist repo locally and in AdoptOpenJDK jenkins and always got:

```
sophiags-MacBook-Pro:antTest sophiag$ git ls-remote -h https://github.com/AdoptOpenJDK/openjdk-jdk19999u.git
Username for 'https://github.com':

jenkins@test-scaleway-ubuntu1604-x64-1:~/workspace/sophiaExternal$ git ls-remote -h https://github.com/AdoptOpenJDK/openjdk-jdk19999u.git
Username for 'https://github.com':

```

Reply:
Updated